pydra/utils/tests/test_hash.py

```diff
@@ -143,11 +158,63 @@ class MyClass:
     assert re.match(rb".*\.MyClass:{str:1:x=.{16}}", obj_repr)
 
 
-def test_bytes_repr_type():
+def test_bytes_repr_attrs_no_slots():
+    @attrs.define(slots=False)
+    class MyClass:
+        x: int
+
+    obj_repr = join_bytes_repr(MyClass(1))
+    assert re.match(rb".*\.MyClass:{str:1:x=.{16}}", obj_repr)
+
+
+def test_bytes_repr_type1():
     obj_repr = join_bytes_repr(Path)
     assert obj_repr == b"type:(pathlib.Path)"
 
 
+def test_bytes_repr_type1a():
+    obj_repr = join_bytes_repr(Zip[Json])
+    assert re.match(rb"type:\(fileformats.application.archive.Json__Zip\)", obj_repr)
+
+
+def test_bytes_repr_type2():
+    T = ty.TypeVar("T")
+
+    class MyClass(ty.Generic[T]):
+        pass
+
+    obj_repr = join_bytes_repr(MyClass[int])
+    assert re.match(rb"type:\(pydra.utils.tests.test_hash.MyClass\[.{16}\]\)", obj_repr)
```

Comment:
If we adopt the proposal above, I believe this would become:

```suggestion
    assert obj_repr == b"type:(pydra.utils.tests.test_hash.MyClass[type:(int)])"
```